metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishedRecordController.java

```diff
@@ -0,0 +1,142 @@
+    final int added = depublishedRecordService
+            .addRecordsToBeDepublished(metisUser, datasetId, recordIdsInSeparateLines);
+    LOGGER.info("{} Depublished records added to dataset with datasetId: {}", added, datasetId);
+  }
+
+  /**
+   * Adds a list of depublished records to the dataset - the version for a multipart file.
+   *
+   * @param authorization the HTTP Authorization header, in the form of a Bearer Access Token.
+   * @param datasetId The dataset ID to which the depublished records belong.
+   * @param recordIdsFile The file containing the record IDs in separate lines.
+   * @throws GenericMetisException which can be one of:
+   * <ul>
+   * <li>{@link NoDatasetFoundException} if the dataset for datasetId was not found.</li>
+   * <li>{@link UserUnauthorizedException} if the user is unauthorized</li>
+   * <li>{@link BadContentException} if some content or the operation were invalid</li>
+   * </ul>
+   */
+  @PostMapping(value = RestEndpoints.DEPUBLISHED_RECORDS_DATASETID, consumes = {
+          MediaType.MULTIPART_FORM_DATA_VALUE})
+  @ResponseStatus(HttpStatus.CREATED)
+  public void createDepublishedRecords(
+          @RequestHeader("Authorization") String authorization,
+          @PathVariable("datasetId") String datasetId,
+          @RequestPart("depublicationFile") MultipartFile recordIdsFile
+  ) throws GenericMetisException, IOException {
+    createDepublishedRecords(authorization, datasetId,
+            new String(recordIdsFile.getBytes(), StandardCharsets.UTF_8));
+  }
+
+  /**
+   * Retrieve the list of depublished records for a specific dataset.
+   *
+   * @param authorization the HTTP Authorization header, in the form of a Bearer Access Token.
+   * @param datasetId The ID of the dataset for which to retrieve the records.
+   * @param page The page to retrieve.
+   * @param sortField The field on which to sort.
+   * @param sortAscending The direction in which to sort.
+   * @param searchQuery Search query for the record ID.
+   * @return A list of records.
+   * @throws GenericMetisException which can be one of:
+   * <ul>
+   * <li>{@link NoDatasetFoundException} if the dataset for datasetId was not found.</li>
+   * <li>{@link UserUnauthorizedException} if the user is unauthorized</li>
+   * </ul>
+   */
+  @GetMapping(value = RestEndpoints.DEPUBLISHED_RECORDS_DATASETID, produces = {
+          MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ResponseListWrapper<DepublishedRecordView> getDepublishedRecords(
+          @RequestHeader("Authorization") String authorization,
+          @PathVariable("datasetId") String datasetId,
+          @RequestParam(value = "page", defaultValue = "0") int page,
+          @RequestParam(value = "sortField", required = false) DepublishedRecordSortField sortField,
+          @RequestParam(value = "sortAscending", defaultValue = "" + true) boolean sortAscending,
```

Comment:
How about the following:

@RequestParam(value = "sortField", required = false, defaultValue = "RECORD_ID") DepublishedRecordSortField sortField,
          @RequestParam(value = "sortAscending", required = false, defaultValue = "true") boolean sortAscending,

Reply:
1. If you put a defaultValue, you don't need required=false anymore.
2. I wanted to prevent having the enum name as a string constant
3. I wanted to ensure that we know that it is a boolean true value that we're expecting.
Your opinion?

Reply:
1. Okay
2. Also okay if that's your preference, don't have a strong opinion here
3. This is similar to you number 2 explanation, so no strong opinion either.

I'll merge then